cpp/src/reductions/simple_segmented.cuh

```diff
@@ -218,6 +227,49 @@ std::unique_ptr<column> string_segmented_reduction(column_view const& col,
   CUDF_FAIL("Segmented reduction on string column only supports min and max reduction.");
 }
 
+/**
+ * @brief Fixed point segmented reduction for 'min', 'max'.
+ *
+ * @tparam InputType    the input column data-type
+ * @tparam Op           the operator of cudf::reduction::op::
+
+ * @param col Input column of data to reduce.
+ * @param offsets Indices to segment boundaries.
+ * @param null_handling If `null_policy::INCLUDE`, all elements in a segment
+ * must be valid for the reduced value to be valid. If `null_policy::EXCLUDE`,
+ * the reduced value is valid if any element in the segment is valid.
+ * @param stream Used for device memory operations and kernel launches.
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return Output column in device memory
+ */
+
+template <typename InputType,
+          typename Op,
+          CUDF_ENABLE_IF(std::is_same_v<Op, cudf::reduction::op::min> ||
+                         std::is_same_v<Op, cudf::reduction::op::max>)>
```

Comment:
Are there other basic operations we could enable "for free" in the same way? Sums don't require adjusting the scale, right?

I can't remember how any/all reductions work. They require `bool8` output type, but can their inputs be decimal in the non-segmented case? I can't see what would prohibit that, from a short investigation, but I haven't tested it.

This is fine to push to a follow-up PR but we should try to enable everything that we can enable for the current complexity of machinery (product, sum of squares would increase the complexity because of the scale shifting logic).

Reply:
I believe any/all are casting the inputs to boolean and compute with min and max. It goes to a different dispatcher (`bool_type_dispatcher` maybe?). Yeah we might be able to enable them for free.

Reply:
Let's punt this to a future PR. Enabling them is fine but we also want good test coverage for them, which might clutter the review process.